[css-contain] Improve references on test suite

mrego opened the pull request:
This patch is just a refactoring to reuse reference files and
name them better. Not having a reference file associated to
a particular test (e.g. contain-size-001) and referenced in many other.

This patch tries to reuse reference files from css/reference/
when possible.
In other cases it renames the reference files in reference/
when reused by several tests to make them generic.

This was suggested by @mstensho in a Chromium review:
https://chromium-review.googlesource.com/c/chromium/src/+/1080799

frivoal commented on css/css-contain/contain-layout-breaks-001.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=assert content="layout containment allows forced breaks.">
-  <link rel="match" href="reference/contain-style-breaks-004-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-two-lines.html">
   <link rel=help href="https://drafts.csswg.org/css-contain-1/#containment-layout">
```
The current naming follows an established convention, documented in the csswg's wiki: https://wiki.csswg.org/test/reftest#the-reftest-reference-file

> References should be named after the earliest test that uses them in the test-topic series they belong to, and must have either -ref or -notref appended to the name
>[...]
> For example, if two self-describing tests list-style-type-003.xht and list-style-type-004.xht share the same reference, that reference could be named list-style-type-003-ref.xht. 

I don't have a strong opinion, but tons of existing tests references are named like that. However, that convention is not documented in https://web-platform-tests.org/ which doesn't have anything to say about reference naming at all, as far as I can tell.

So, should we preserve that convention (and if so, document it on web-platform-tests.org)? Should we replace it with something else (and document that)?

CCing @gsnedders @foolip @fantasai @svgeesus for opinion

mrego replied:
Ok, I didn't know about that convention.
The documentation in https://web-platform-tests.org/writing-tests/reftests.html says:
> For references that are applicable only to a single test, it is recommended to use the test name with a suffix of -ref as their filename; e.g., test.html would have test-ref.html as a reference.

But it doesn't mention what happens when more tests start to use it.

I believe @mstensho has a point that having a test `contain-layout-005` pointing to `contain-size-001-ref` is somehow confusing, but I don't have a strong opinion on this topic either.

I can left these changes out of the PR, and we can discuss in an issue if needed.
I believe the changes reusing the the references from `css/reference/` are welcomed by everyone, aren't they?

foolip replied:
Just an opinion: when there's a number in the ref file and a second test is added using the same ref, rename the ref to something more generic like the ref-filled-green-200px-square.html files in css/reference/

gsnedders replied:
I agree with the attributed-to-@mstensho view that pointing to other numbered refs is confusing.

On the whole, I think filename rules aren't that beneficial (in general, it's probably better to try and make the file name descriptive than adding an incrementing number onto it!).

mrego replied:
@frivoal are you planing to review this again? Or what should we do move this forward?

I can split this in 2 PRs:
* One for using generic references in the current tests when possible.
* Another one for changing the names, once there's agreement about that being the right thing todo.

mrego replied:
> * One for using generic references in the current tests when possible.

This has been done in #11667. Once that's merged I'd update this PR with the other part.

frivoal commented on css/css-contain/contain-paint-007.html:
```diff
@@ -5,12 +5,11 @@
   <meta name=flags content="">
   <meta name=assert content="paint containment does not apply to ruby-text-container">
-  <link rel="match" href="reference/contain-paint-007-ref.html">
+  <link rel="match" href="reference/contain-rtc-pass.html">
```
Same comment as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-size-004.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="size containment does not to apply ruby-text-container, which is an internatl ruby element">
-  <link rel="match" href="reference/contain-size-004-ref.html">
+  <link rel="match" href="reference/contain-rtc-pass.html">
```
same comment as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-breaks-001.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment is (no longer) supposed to have any effect on the break-inside property">
-  <link rel="match" href="reference/contain-style-breaks-001-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-four-lines.html">
```
same comment as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-breaks-002.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment is not (any longer) supposed to have any effect on the break-inside property. Same as -001, applying containment on the parent.">
-  <link rel="match" href="reference/contain-style-breaks-001-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-four-lines.html">
```
same comment as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-breaks-003.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment is not (any longer) supposed to have any effect on the break-inside property. Same as -001, applying break-inside on the parent.">
-  <link rel="match" href="reference/contain-style-breaks-001-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-four-lines.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-breaks-004.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment is not (any longer) supposed to have any effect on the break-before property.">
-  <link rel="match" href="reference/contain-style-breaks-004-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-two-lines.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-breaks-005.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment is not (any longer) supposed to have any effect on the break-after property.">
-  <link rel="match" href="reference/contain-style-breaks-004-ref.html">
+  <link rel="match" href="reference/contain-two-blocks-a-letters-two-lines.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/contain-style-counters.html:
```diff
@@ -1,7 +1,7 @@
 <title>CSS Containment Test: contain:style for counters</title>
 <link rel="help" href="https://drafts.csswg.org/css-contain-1/#containment-style">
-<link rel="match" href="contain-style-counters-ref.html">
+<link rel="match" href="reference/contain-style-counters-ref.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/counter-scoping-001.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="counter-increment is scoped to the subtree and creates a new counter at the root of the subtree">
-  <link rel="match" href="reference/counter-scoping-001-ref.html">
+  <link rel="match" href="reference/counter-scoping-1-1.2.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/counter-scoping-002.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="counter-set is scoped to the subtree and creates a new counter at the root of the subtree">
-  <link rel="match" href="reference/counter-scoping-001-ref.html">
+  <link rel="match" href="reference/counter-scoping-1-1.2.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/quote-scoping-003.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment cause the no-open-quote value of the content property are scoped to the element's subtree">
-  <link rel="match" href="reference/quote-scoping-003-ref.html">
+  <link rel="match" href="reference/quote-scoping-az.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

frivoal commented on css/css-contain/quote-scoping-004.html:
```diff
@@ -5,7 +5,7 @@
   <meta name=flags content="">
   <meta name=assert content="style containment cause the no-close-quote value of the content property are scoped to the element's subtree">
-  <link rel="match" href="reference/quote-scoping-003-ref.html">
+  <link rel="match" href="reference/quote-scoping-az.html">
```
same as https://github.com/web-platform-tests/wpt/pull/11323/files#r192678192

mrego committed:
[css-contain] Improve reference naming on test suite

Instead of using a reference associated to a particular test
in several tests, this patch renames the references
to a more generic name.

This was suggested by @mstensho in a Chromium review:
https://chromium-review.googlesource.com/c/chromium/src/+/1080799

mrego commented:
Afert #11667 has been merged, this is now much smaller, I've pushed a new rebased version.

@frivoal feel free to close it without merging if you think it's not completely clear what we should do here.

frivoal commented:
There is existing documentation that says that refs should be named after the first test that they match. That documentation is a old, and in the csswg wiki. It's unclear if it is supposed to apply to wpt, but at the same time, wpt does not document any other practice, so presumably it still applies, at least for css.

I want either:
* agreement that the rules haven't changed from the old csswg practices, document that in wpt, and reject this PR.
* agreement that using descriptive names for reused refs is better, document that in wpt, delete the contradicting information from the csswg wiki, and accept this PR.

I don't care which one we do, as long as we keep documentation and practice in sync.

Several people in this thread have argue in favor of using descriptive names for shared refs is better, so I guess we should go with that.

I would still like to hear from someone who was involved in setting the previous policy before overturning it though. @plinss @fantasai @svgeesus

frivoal commented:
@gsnedders I've opened https://github.com/web-platform-tests/wpt/issues/12484 to discuss this. Let's come back to this PR once that is settled.
